frontend/appflowy_flutter/lib/user/application/auth/appflowy_auth_service.dart

```diff
@@ -0,0 +1,65 @@
+import 'package:appflowy/user/application/auth/auth_service.dart';
+import 'package:appflowy/user/application/user_service.dart';
+import 'package:dartz/dartz.dart';
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flowy_infra/uuid.dart';
+import 'package:appflowy_backend/dispatch/dispatch.dart';
+import 'package:appflowy_backend/protobuf/flowy-error/errors.pb.dart';
+import 'package:appflowy_backend/protobuf/flowy-user/protobuf.dart'
+    show SignInPayloadPB, SignUpPayloadPB, UserProfilePB;
+
+import '../../../generated/locale_keys.g.dart';
+
+class AppFlowyAuthService implements AuthService {
+  @override
+  Future<Either<FlowyError, UserProfilePB>> signIn({
+    required String email,
+    required String password,
+  }) async {
+    //
+    final request = SignInPayloadPB.create()
+      ..email = email
+      ..password = password;
+
+    final response = UserEventSignIn(request).send();
+    return response.then((value) => value.swap());
```

Comment:
```suggestion
    final response = await UserEventSignIn(request).send();
    return response.swap());
```

Reply:
both of them look good to me. `await` or `then`. 

Sure, we can generate the swap function while converting the PB event to a function.